src/main/java/ru/sbtqa/tag/pagefactory/drivers/TagWebDriver.java

```diff
@@ -137,11 +136,66 @@ private static void createDriver() throws UnsupportedBrowserException, Malformed
         webDriver.get(WEBDRIVER_STARTING_URL);
     }
 
+    private static void configureDriver(BrowserManager webDriverManager, String browserType) {
+        final String noDriverPathWarnMessage = "The value of property 'webdriver.drivers.path is not specified."
+                + " Trying to automatically download and setup driver.";
+
+        if (!WEBDRIVER_PATH.isEmpty()) {
+            System.setProperty("webdriver." + browserType + ".driver", new File(WEBDRIVER_PATH).getAbsolutePath());
+        } else {
+            LOG.warn(noDriverPathWarnMessage, WEBDRIVER_BROWSER_NAME);
+            configureWebDriverManagerParams(webDriverManager, browserType);
+            webDriverManager.setup();
+        }
+    }
+
+    private static void configureProxy(DesiredCapabilities capabilities) {
+        if (!WEBDRIVER_PROXY.isEmpty()) {
+            setProxy(new BrowserMobProxyServer());
+            proxy.start(0);
+            Proxy seleniumProxy = ClientUtil.createSeleniumProxy(proxy);
+            capabilities.setCapability(CapabilityType.PROXY, seleniumProxy);
+        }
+    }
+
+    private static void configureWebDriverManagerParams(BrowserManager webDriverManager, String browserType) {
+        if (!WEBDRIVER_BROWSER_VERSION.isEmpty() && !browserType.equalsIgnoreCase(BrowserType.IE)) {
+            LOG.info("You have specified 'webdriver.browser.version' property. Trying to find corresponding driver.");
+
+            String mappedVersion = parseDriverVersionFromMapping(WEBDRIVER_BROWSER_VERSION, browserType.toLowerCase());
+            webDriverManager.version(mappedVersion);
```

Comment:
NPE? mappedVersion can be null

Reply:
I've checked. BrowserManager.version(null) causes no NPE